pkg/controller/controller_utils.go

```diff
@@ -525,6 +525,43 @@ func (f *FakePodControl) Clear() {
 	f.Templates = []api.PodTemplateSpec{}
 }
 
+// ByLogging allows custom sorting of pods so the best one can be picked for getting its logs.
+type ByLogging []*api.Pod
+
+func (s ByLogging) Len() int      { return len(s) }
+func (s ByLogging) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
+
+func (s ByLogging) Less(i, j int) bool {
+	// 1. assigned < unassigned
+	if s[i].Spec.NodeName != s[j].Spec.NodeName && (len(s[i].Spec.NodeName) == 0 || len(s[j].Spec.NodeName) == 0) {
+		return len(s[i].Spec.NodeName) > 0
+	}
+	// 2. PodRunning < PodUnknown < PodPending
```

Comment:
its really weird to see this and then have you make the numbers opposite the comment.